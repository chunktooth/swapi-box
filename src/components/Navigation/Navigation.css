.Navigation {
	position: relative;
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	top: 240px;
	margin: auto;
}

.nav-btn {
	background-color: hsla(0, 0%, 0%, .7);
	color: #92ACA3;
	width: 160px;
	margin: 20px;
	border: 2px solid #2B3536;
	border-radius: 13px;
	box-shadow: 0px 0px 20px 2px #0ff;
	font-size: 1.2em;
	font-family: 'Quantico', sans-serif;
}

.nav-btn:hover {
	box-shadow: 0px 0px 50px 2px #0ff;
}

@media screen and (max-width: 700px) {
	.nav-btn {
		width: 150px;
	}
}

@media screen and (max-width: 600px) {
	.Navigation {
		flex-direction: column;
		top: 310px;
	}

	.nav-btn {
		width: 78%;
	}
}