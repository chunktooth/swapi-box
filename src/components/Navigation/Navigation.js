import React from 'react';
import {NavLink} from 'react-router-dom';
import PropTypes from 'prop-types';
import './Navigation.css';

const Navigation = ({setCategory}) => {
	return (
		<section className="Navigation">
			
			<NavLink to='/people' className='nav'>
				<button 
					className='nav-btn'
					onClick={() => setCategory('people')}>
					PEOPLE
				</button>
			</NavLink>

			<NavLink to='/planets' className='nav'>
				<button 
					className='nav-btn'
					onClick={() => setCategory('planets')}>
					PLANETS
				</button>
			</NavLink>

			<NavLink to='/vehicles' className='nav'>
				<button 
					className='nav-btn'
					onClick={() => setCategory('vehicles')}>
					VEHICLES
				</button>
			</NavLink>
			
		</section>
	)
}

export default Navigation;




