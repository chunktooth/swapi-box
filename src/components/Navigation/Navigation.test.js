import React from 'react';
import Navigation from './Navigation';
import {shallow} from 'enzyme';

describe('Navigation', () => {

	it('should match the snapshot', () => {
		const wrapper = shallow(<Navigation />);	
		expect(wrapper).toMatchSnapshot();
	});

	it('should set the right category after clicking a category button', () => {
		const mockDataType = "people";
		const mockFunction = jest.fn();
		const wrapper = shallow(<Navigation
			data={mockDataType} 
			setCategory={mockFunction}
		/>);	
		wrapper.find('button').first().simulate('click');
		expect(mockFunction).toHaveBeenCalledWith(mockDataType);
	});

});