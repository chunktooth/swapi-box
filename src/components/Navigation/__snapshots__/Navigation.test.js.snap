// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    setCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/people"
>
          <button
                    className="nav-btn"
                    onClick={[Function]}
          >
                    PEOPLE
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/planets"
>
          <button
                    className="nav-btn"
                    onClick={[Function]}
          >
                    PLANETS
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/vehicles"
>
          <button
                    className="nav-btn"
                    onClick={[Function]}
          >
                    VEHICLES
          </button>
</NavLink>,
      ],
      "className": "Navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="nav-btn"
            onClick={[Function]}
>
            PEOPLE
</button>,
          "className": "nav",
          "to": "/people",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PEOPLE",
            "className": "nav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PEOPLE",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="nav-btn"
            onClick={[Function]}
>
            PLANETS
</button>,
          "className": "nav",
          "to": "/planets",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PLANETS",
            "className": "nav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PLANETS",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="nav-btn"
            onClick={[Function]}
>
            VEHICLES
</button>,
          "className": "nav",
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "VEHICLES",
            "className": "nav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "VEHICLES",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/people"
>
            <button
                        className="nav-btn"
                        onClick={[Function]}
            >
                        PEOPLE
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/planets"
>
            <button
                        className="nav-btn"
                        onClick={[Function]}
            >
                        PLANETS
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/vehicles"
>
            <button
                        className="nav-btn"
                        onClick={[Function]}
            >
                        VEHICLES
            </button>
</NavLink>,
        ],
        "className": "Navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="nav-btn"
              onClick={[Function]}
>
              PEOPLE
</button>,
            "className": "nav",
            "to": "/people",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEOPLE",
              "className": "nav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "PEOPLE",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="nav-btn"
              onClick={[Function]}
>
              PLANETS
</button>,
            "className": "nav",
            "to": "/planets",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PLANETS",
              "className": "nav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "PLANETS",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="nav-btn"
              onClick={[Function]}
>
              VEHICLES
</button>,
            "className": "nav",
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "VEHICLES",
              "className": "nav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "VEHICLES",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
